simd/src/Kokkos_SIMD_AVX512.hpp

```diff
@@ -86,7 +86,7 @@ class simd_mask<T, simd_abi::avx512_fixed_size<8>> {
   }
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION value_type
   operator[](std::size_t i) const {
-    return static_cast<value_type>(reference(m_value, int(i)));
+    return reference(const_cast<simd_mask*>(this)->m_value, int(i));
```

Comment:
Should we just have a const_reference type which takes a const value reference as a parameter in its constructor? I know its more code but I do REALLY dislike casting const away from values. I actually in some ways want to argue for a compiler to have a "const means const" mode, where you can't const_cast and compilers don't have to assume non-const things alias const things.

Reply:
I second this, I really don't like `const_cast`

Reply:
Sure! We don't even need the type really because it's just converted to `bool` immediately so I pushed a change to do this